plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java

```diff
@@ -19,107 +19,201 @@
 
 import org.eclipse.jface.dialogs.Dialog;
 import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.resource.JFaceResources;
 import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.CLabel;
 import org.eclipse.swt.events.ModifyListener;
 import org.eclipse.swt.events.SelectionAdapter;
 import org.eclipse.swt.events.SelectionEvent;
 import org.eclipse.swt.layout.GridData;
 import org.eclipse.swt.layout.GridLayout;
 import org.eclipse.swt.widgets.*;
+import org.jkiss.code.NotNull;
+import org.jkiss.code.Nullable;
+import org.jkiss.dbeaver.Log;
 import org.jkiss.dbeaver.model.connection.DBPDriverLibrary;
 import org.jkiss.dbeaver.registry.driver.DriverDescriptor;
 import org.jkiss.dbeaver.registry.driver.DriverLibraryMavenArtifact;
 import org.jkiss.dbeaver.registry.maven.MavenArtifactReference;
 import org.jkiss.dbeaver.ui.UIUtils;
 import org.jkiss.dbeaver.ui.internal.UIConnectionMessages;
 import org.jkiss.utils.CommonUtils;
+import org.jkiss.utils.xml.SAXListener;
+import org.jkiss.utils.xml.SAXReader;
+import org.jkiss.utils.xml.XMLException;
+import org.xml.sax.Attributes;
 
-/**
- * EditMavenArtifactDialog
- */
-class EditMavenArtifactDialog extends Dialog
-{
-    private DriverLibraryMavenArtifact library;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class EditMavenArtifactDialog extends Dialog {
+    private static final Log log = Log.getLog(EditMavenArtifactDialog.class);
+
+    private static final Pattern REGEX_FOR_GRADLE = Pattern.compile("^\\s*(?:(group:\\s+'(?<group>.*?)',\\s+name:\\s+'(?<name>.*?)',\\s+version:\\s+'(?<version>.*?)')|'(?<shortGroup>.*?):(?<shortName>.*?):(?<shortVersion>.*?)')\\s*$", Pattern.MULTILINE);
+
+    private final DriverLibraryMavenArtifact originalArtifact;
+    private final List<DriverLibraryMavenArtifact> artifacts = new ArrayList<>();
+    private boolean ignoreDependencies;
+    private CLabel errorLabel;
+    private boolean loadOptionalDependencies;
+    private final DriverDescriptor driver;
     private Text groupText;
     private Text artifactText;
     private Text classifierText;
     private Combo versionText;
-    private boolean ignoreDependencies;
-    private boolean loadOptionalDependencies;
+    private Text fieldText;
+    private boolean manualSelected;
```

Comment:
Please reorder these fields to be more pleasant to look at. Finals on top, then other variables, SWT components afterwards.